proposal/chrisPyr/README.rst

```diff
@@ -0,0 +1,118 @@
+
+| We all learned "Kepler's laws of planetary motion"[1] in high school.
+| The Kepler's laws can be explained and proved by Newtonian Mechanics[2].
+| Though the laws would fail to predict the orbits or speed of planets accurately in some certain conditions, e.g. Perihelion Motion of Mercury[3](fig. 1) and that's why we need "Theory of Relativity" to help us to describe and predict more precisely.
+| Einstein did prove this on his paper[4].
+| But that's not a point I need to worry about, Newtonian Mechanics should predict precise enough. I plan to build up a program that can predict the orbits of planets only take Newtonian Mechanics into account.
+| The kind of problem is actually a N-Body problem. And there is no general solution to N-Body problem, we can only try to simulate it with a numerical method.
+| The simplest idea is to compute all interation force between each body, which is a Brute-force method.
+| By calculating the acceleration and velocity of each time step, we can predict the position of bodies next moment. However, in real life the acceleration and velocity changes even in each time step.
+| Although when time step is small enough, the results should be good enough. But we can try to use "Runge-Kuta Methods"[5] to improve results.
+| In my plan, the final product should be able to show the orbits, periods of planets. And it can be compared with the real world data to prove the correctness and also incorrectness of Newtonian Mechanics.
+| And also do I want to add some intersting feature into it, e.g. users can add/delete planets into/in the system to see how the entire system changes.
+
+
+.. image:: images/Precessing_Kepler_orbit_280frames_e0.6_smaller.gif
+
+Prospective Users
+=================
+
+| This can be used as a supplementary tutorial to high school students and a example to N-body simulation.
+
+System Architecture
+===================
+
+| The default of building the system would create the solar system in a gif or mp4 format. I plan to use Python matplotlib to do the animatin creation part[6].
+| Users can add one or more planets by providing three parameters: intial velocity, position, and mass.
+| To build the solar system I also need information about solar system initial parameters, I plan to get the data from "JPL Horizon(NASA)"[7].
+
+Needed Parameters:
+    - default(build a solar system)
+    - initial acceleration, velocity, position
+    - time step, time period(tunable)
+
+Program Features:
+    - export information of calculated result such as formula of orbit, speed/period of a planet
+    - export a gif or mp4 file to show the drawing result within a time period
+    - add user-defined planets
+    - delete planets
+
+Constraints:
+    - only be able to simulate a period of time, because N-body problem is a chaos system, we don't know the trajectory after our simulation
+    - we cannot perform real integration on computer, so the results may not be perfectly match the real world
```

Comment:
What we call "numerical integration" is actually quadrature.  It may take a couple of weeks to discuss in a numerical analysis course.